lxd/certificates.go

```diff
@@ -183,16 +219,26 @@ func certificatesPost(d *Daemon, r *http.Request) Response {
 	return EmptySyncResponse
 }
 
-var certificatesCmd = Command{"certificates", false, true, certificatesGet, nil, certificatesPost, nil}
+var certificatesCmd = Command{
+	"certificates",
+	false,
+	true,
+	certificatesGet,
+	nil,
+	certificatesPost,
+	nil,
+}
 
 func certificateFingerprintGet(d *Daemon, r *http.Request) Response {
 	fingerprint := mux.Vars(r)["fingerprint"]
 
 	for _, cert := range d.clientCerts {
-		if fingerprint == shared.GenerateFingerprint(&cert) {
-			b64 := base64.StdEncoding.EncodeToString(cert.Raw)
-			body := shared.Jmap{"type": "client", "certificates": b64}
-			return SyncResponse(true, body)
+		if strings.HasSuffix(shared.GenerateFingerprint(&cert), fingerprint) {
+			resp := certificateResponse{}
+			resp.Fingerprint = shared.GenerateFingerprint(&cert)
+			resp.Certificate = base64.StdEncoding.EncodeToString(cert.Raw)
+			resp.Type = "client"
+			return SyncResponse(true, resp)
 		}
```

Comment:
I wonder if it wouldn't be better to match what we do for images there and go hit the database, that way we can do a LIKE query and ensure that we only have a single match.